src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -34,11 +38,14 @@ public final class KeyContainsMasker implements JsonMasker {
      * @param maskingConfig the masking configurations for the created masker
      */
     public KeyContainsMasker(JsonMaskingConfig maskingConfig) {
-        this.allowMode = maskingConfig.getTargetKeyMode() == JsonMaskingConfig.TargetKeyMode.ALLOW;
         this.maskingConfig = maskingConfig;
         this.targetKeysTrie = new ByteTrie(!maskingConfig.caseSensitiveTargetKeys());
         for (String key : maskingConfig.getTargetKeys()) {
-            this.targetKeysTrie.insert(key);
+            if (maskingConfig.caseSensitiveTargetKeys()) {
+                this.targetKeysTrie.insert(key);
+            } else {
+                this.targetKeysTrie.insert(key.toLowerCase());
```

Comment:
This shouldn't be necessary - trie already converts to lowercase/uppercase on prowler character basis, did it not work?

Reply:
@gavlyukovskiy I think you are right, I just did this to replace:

```java
        if (!caseSensitiveTargetKeys) {
            targets = targets.stream().map(String::toLowerCase).collect(Collectors.toSet());
        }
        targetKeys = targets;
 ```
        
        Probably you forgot to remove that while introducing ByteTrie?